src/imports.rs

```diff
@@ -250,13 +250,29 @@ impl<'a> FmtVisitor<'a> {
         self.last_pos = pos_after_last_use_item;
     }
 
-    pub fn format_import(&mut self, vis: &ast::Visibility, vp: &ast::ViewPath, span: Span) {
+    pub fn format_import(
+        &mut self,
+        vis: &ast::Visibility,
+        vp: &ast::ViewPath,
+        span: Span,
+        attrs: &[ast::Attribute],
+    ) {
         let vis = utils::format_visibility(vis);
         // 4 = `use `, 1 = `;`
         let rw = Shape::indented(self.block_indent, self.config)
             .offset_left(vis.len() + 4)
             .and_then(|shape| shape.sub_width(1))
-            .and_then(|shape| vp.rewrite(&self.get_context(), shape));
+            .and_then(|shape| match vp.node {
```

Comment:
I would try to avoid adding anything to `RewriteContext`, it is tempting, and usually a bad idea, to make this kind of context a dumping ground for node-specific state. I think the solution here is essentially the right design...